### ISSUE TITLE
[Backport] Nodes fail to update, with cluster stuck in Upgrading state when attempting to update private registry details in Rancher provisioned cluster

### ISSUE THREAD

**Issue by Jono-SUSE-Rancher** [rancher/rancher#33697]
We need to backport rancher/rancher#30628 to v2.5.10.

**Comment by rawmind0** [comment]
Already done https://github.com/rancher/rancher/pull/33009, PR is already merged at release/v2.5

**Comment by slickwarren** [comment]
on v2.5.10-rc1, the downstream cluster is now able to update after following instructions from the parent ticket:
* Provision a Rancher instance (I tested with v2.4.0, v2.4.8 and v2.5.3) and a custom cluster with 3 all role nodes and 3 worker nodes. I used https://github.com/superseb/tf-do-rancher2
* Once the custom cluster is successfully provisioned edit the cluster to add URL, username and password for a valid private registry with the applicable set of images.

-> all nodes are updated and are using the private registry

Reopening for the following concerns: 
* the `This cluster is currently Upgrading.` warning in the `cluster` view no longer appears at all after making the private registry update. I did a quick check for other updates, and the warning does appear for those.  Can we still show this warning while the private registry images are being applied to the cluster?
